Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp

```diff
@@ -229,6 +231,48 @@ bool getSampleVideoInfo(GstSample* sample, GstVideoInfo& videoInfo)
 }
 #endif
 
+std::optional<TrackID> getTrackIdFromPad(const GRefPtr<GstPad>& pad)
+{
+    const gchar* streamIdAsCharacters = gst_pad_get_stream_id(pad.get());
```

Comment:
This leaks, `gst_pad_get_stream_id()` returns a newly allocated string.

Reply:
Thats oddly inconsistent since `gst_stream_get_stream_id()` doesnt, but thanks for pointing it out!